stratisd_optional.py

```diff
@@ -0,0 +1,418 @@
+    :return: str fsdevdest, str fsdevmapperlinkdest
+    """
+    objects = StratisDbus.get_managed_objects()
+
+    pool_gmodata = objects[pool_path]
+    pool_uuid = pool_gmodata[StratisDbus.POOL_IFACE]["Uuid"]
+    filesystem_gmodata = objects[filesystem_path]
+    filesystem_uuid = filesystem_gmodata[StratisDbus.FS_IFACE]["Uuid"]
+
+    filesystem_devnode = "/dev/stratis/" + pool_name + "/" + filesystem_name
+
+    fs_devmapperlinkstr = (
+        "/dev/mapper/stratis-1-" + pool_uuid + "-thin-fs-" + filesystem_uuid
+    )
+
+    fsdevdest = resolve_symlink(filesystem_devnode)
+    fsdevmapperlinkdest = resolve_symlink(fs_devmapperlinkstr)
+    return fsdevdest, fsdevmapperlinkdest
+
+
+class StratisCertify(unittest.TestCase):
+    """
+    Unit tests for the stratisd package.
+    """
+
+    def _inequality_test(self, result, expected_non_result):
+        """
+        :param object result: the result of a test
+        :param object expected_non_result: a value which the result must
+                                           not match, but which has the
+                                           expected type
+        """
+        self.assertIsInstance(result, type(expected_non_result))
+        self.assertNotEqual(result, expected_non_result)
+
+    def _unittest_command(self, result, expected_return_code):
+        """
+        :param result: a tuple of the (optional) return value, the
+                       return code, and the return message from a
+                       D-Bus call
+        :type result: tuple of object * dbus.UInt16 * str OR tuple
+                      of dbus.UInt16 * str if there is no return value
+        :raises: AssertionError if the actual return code is not
+                 equal to the expected return code
+        """
+        if len(result) == 3:
+            (_, return_code, msg) = result
+        else:
+            (return_code, msg) = result
+
+        self.assertEqual(return_code, expected_return_code, msg=msg)
+
+
+class StratisdCertify(StratisCertify):  # pylint: disable=too-many-public-methods
+    """
+    Tests on stratisd, the principal daemon.
+    """
+
+    def setUp(self):
+        """
+        Setup for an individual test.
+        * Register a cleanup action, to be run if the test fails.
+        * Ensure that stratisd is running via systemd.
+        * Use the running stratisd instance to destroy any existing
+        Stratis filesystems, pools, etc.
+        * Call "udevadm settle" so udev database can be updated with changes
+        to Stratis devices.
+        :return: None
+        """
+        self.addCleanup(clean_up)
+
+        if process_exists("stratisd") is None:
+            exec_command(["systemctl", "start", "stratisd"])
+            time.sleep(20)
+
+        clean_up()
+
+        time.sleep(1)
+        exec_command(["udevadm", "settle"])
+
+        trace = subprocess.Popen(
+            [
+                "./scripts/monitor_dbus_signals.py",
```

Comment:
We should create a constant for this string, since we use it in two places.

Reply:
Resolved.